static/scripts/rewards/render-transaction/supabase-getters.ts

```diff
@@ -0,0 +1,155 @@
+import { createClient } from "@supabase/supabase-js";
+import { Database, type PermitReward } from "@ubiquity-os/permit-generation";
+import { TokenType } from "@ubiquibot/permit-generation/types";
+import { toaster } from "../toaster";
+import { BigNumber } from "ethers";
+
+declare const SUPABASE_URL: string;
+declare const SUPABASE_ANON_KEY: string;
+export const supabase = createClient<Database>(SUPABASE_URL, SUPABASE_ANON_KEY);
+
+export interface SupabasePermit {
+  id: number;
+  nonce: string;
+  amount: string;
+  deadline: string;
+  signature: string;
+  transaction: string | null;
+  token_id: number | null;
+  beneficiary_id: number | null;
+  partner_id: number | null;
+}
+
+export interface SupabaseToken {
+  id: number;
+  address: string;
+  network: number;
+}
+
+export interface SupabasePartner {
+  id: number;
+  wallet_id: number;
+}
+
+export interface SupabaseUser {
+  id: number;
+  wallet_id: number;
+}
+
+export interface SupabaseWallet {
+  id: number;
+  address: string;
+}
+
+export async function fetchTokenById(tokenId: number): Promise<SupabaseToken | null> {
+  const { data, error } = await supabase.from("tokens").select("*").eq("id", tokenId).single();
+  if (error) {
+    console.error("error fetching token:", error);
+    return null;
+  }
+  return data as SupabaseToken;
+}
+
+export async function fetchUserById(userId: number): Promise<SupabaseUser | null> {
+  const { data, error } = await supabase.from("users").select("*").eq("id", userId).single();
+  if (error) {
+    console.error("error fetching user:", error);
+    return null;
+  }
+  return data as SupabaseUser;
+}
+
+export async function fetchPartnerById(partnerId: number): Promise<SupabasePartner | null> {
+  const { data, error } = await supabase.from("partners").select("*").eq("id", partnerId).single();
+  if (error) {
+    console.error("error fetching partner:", error);
+    return null;
+  }
+  return data as SupabasePartner;
+}
+
+export async function fetchWalletById(walletId: number): Promise<SupabaseWallet | null> {
+  const { data, error } = await supabase.from("wallets").select("*").eq("id", walletId).single();
+  if (error) {
+    console.error("error fetching wallet:", error);
+    return null;
+  }
+  return data as SupabaseWallet;
+}
+
+export async function fetchPermitsFromSupabase(userId: number): Promise<PermitReward[]> {
+  const { data, error } = await supabase.from("permits").select("*").eq("beneficiary_id", userId);
```

Comment:
add order by.  For me permits were mixed with claimed permits, and it appeared to be a random order. Show newer first maybe?